### PR TITLE
feat: add bitcoin websocket support

### PR DESCRIPTION
**Pull request by kaladinlight**

### CODE REVIEW

File: packages/types/src/chain-adapters/index.ts [hunk 1]
```diff
@@ -104,14 +104,15 @@ export type TxFee = {
 }
 
 export enum TxType {
-  send = 'send',
-  receive = 'receive'
+  Send = 'send',
```
Comment (kaladinlight) [inline packages/types/src/chain-adapters/index.ts:107]: I will be making associated PR for this and bitcoin websockets

**Comment by 0xdef1cafe** [comment]
looks like this needs a new version of unchained client

**Review by elmutt — APPROVED** [review]

File: packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts [hunk 2]
```diff
@@ -339,7 +344,60 @@ export class ChainAdapter implements IChainAdapter<ChainTypes.Bitcoin> {
...
+        const status =
+          msg.confirmations > 0 ? chainAdapters.TxStatus.Confirmed : chainAdapters.TxStatus.Pending
+
+        const baseTx = {
+          address: msg.address,
+          asset: ChainTypes.Bitcoin,
+          blockHash: msg.blockHash,
+          blockHeight: msg.blockHeight,
+          blockTime: msg.blockTime,
+          chain: ChainTypes.Bitcoin as ChainTypes.Bitcoin,
```
Comment (0xdef1cafe) [inline packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts:374]: ?

**Review by 0xdef1cafe — APPROVED** [review]

File: packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts (hunk 2 above)
Comment (kaladinlight) [reply packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts:374]: fixed this type error:
```
Types of property 'chain' are incompatible.
      Type 'ChainTypes' is not assignable to type 'ChainTypes.Bitcoin'.
```

Comment (kaladinlight) [reply packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts:374]: I can type `baseTx` to get around it as well, but this was less lines, is uggo though..

**Pull request merged**